### PR TITLE
Adding JQ Syntax to vim!

### PR DESCRIPTION
**Pull request by vito-c**
I am adding jq syntax to vim as well as updating the maintainer for json.
@itchyny @dtolnay @Freed-Wu

### CODE REVIEW

File: runtime/compiler/jq.vim [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+if exists('b:current_compiler')
+  finish
+endif
+let b:current_compiler = 'jq'
+
+let s:save_cpoptions = &cpoptions
+set cpoptions&vim
+
+if exists(':CompilerSet') != 2
```
Comment (chrisbra) [inline runtime/compiler/jq.vim]: You can remove the `exists(':CompilerSet')` test. We just [removed](https://github.com/vim/vim/commit/408281e16a36c15eed10fbf0406fa8) this from the other compilers.

File: runtime/ftplugin/jq.vim [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+" Last Change:	2024 Apr 17
+" Upstream: https://github.com/vito-c/jq.vim
+
+if exists('b:did_ftplugin')
+  finish
+endif
+let b:did_ftplugin = 1
+
+let s:save_cpoptions = &cpoptions
+set cpoptions&vim
```
Comment (chrisbra) [inline runtime/ftplugin/jq.vim]: You are not using line-continuation here, so you can remove the save and restore of 'cpoptions'

File: runtime/syntax/jq.vim [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+syn keyword jqError _assign _flatten _modify _nwise _plus _negate _minus _multiply
+syn keyword jqError _divide _mod _strindices _equal _notequal _less _greater _lesseq
+syn keyword jqError _greatereq _sort_by_impl _group_by_impl _min_by_impl _max_by_impl _match_impl _input
+" TODO: these errors should show up when doing def _flatten: as well
+
+" Numbers
+syn match jqNumber /\<0[dDfFlL]\?\>/ " Just a bare 0
+syn match jqNumber /\<[1-9]\d*[dDfFlL]\?\>/  " A multi-digit number - octal numbers with leading 0's are deprecated in Scala
+
+if !exists('jq_quote_highlight')
```
Comment (chrisbra) [inline runtime/syntax/jq.vim:110]: Can you add a section at `:h syntax.txt` for this configuration settig?

File: runtime/syntax/jq.vim [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+hi def link jqMacro                  Macro
+hi def link jqError                  Error
+hi def link jqString                 String
+hi def link jqInterpolationDelimiter Delimiter
+"hi def link jqStatement              Statement
+hi def link jqConditional            Conditional
+"hi def link jqRepeat                 Repeat
+"hi def link jqException              Exception
+"hi def link jqInclude                Include
+"hi def link jqDecorator              Define
```
Comment (chrisbra) [inline runtime/syntax/jq.vim]: can you remove those commented out sections?

**Comment by chrisbra** [comment]
can you please also update https://github.com/vim/vim/blob/master/.github/CODEOWNERS?

**Comment by vito-c** [comment]
@chrisbra Should I resolve the comments as I fix the issues or do you want to resolve them after you have verified?

**Comment by chrisbra** [comment]
Ah thanks. The documentation is still wrong, it uses `:set`, but I can fix this when merging.

File: runtime/ftplugin/jq.vim [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  finish
+endif
+let b:did_ftplugin = 1
+
+let b:undo_ftplugin = 'setl commentstring<'
+
+setlocal commentstring=#%s
+compiler jq
+
+let &cpoptions = s:save_cpoptions
```
Comment (vito-c) [inline runtime/ftplugin/jq.vim:17]: I need to remove this as well

**Pull request closed without merging**